### PR TITLE
Facts statcom

### PR DESCRIPTION
**Pull request by rbolgaryn**

### CODE REVIEW

File: pandapower/pypower/idx_ssc.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 1996-2015 PSERC. All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
```
Comment (SteffenMeinecke) [inline pandapower/pypower/idx_ssc.py:5]: Is the copyright, similar to other pypower copyrights, needed for SSC indices, too?

File: pandapower/pf/create_jacobian_facts.py [hunk 2]
```diff
@@ -184,3 +183,376 @@ def create_J_modification_tcsc(V, Ybus_tcsc, x_control, svc_controllable, tcsc_c
     J_m = csr_matrix(J_m)
 
     return J_m
+
+#
+# def create_J_modification_ssc(V, Ybus_ssc, x_control,
```
Comment (SteffenMeinecke) [inline pandapower/pf/create_jacobian_facts.py:188]: Should we keep this old part as a comment?

File: pandapower/pypower/newtonpf.py [hunk 3]
```diff
@@ -266,10 +297,11 @@ def newtonpf(Ybus, Sbus, V0, ref, pv, pq, ppci, options, makeYbus=None):
         if tdpf:
             # update the R, g, b for the tdpf_lines, and the Y-matrices
             branch[tdpf_lines, BR_R] = r = r_ref_pu * (1 + alpha_pu * (T - t_ref_pu))
-            Ybus, Yf, Yt = makeYbus(baseMVA, bus, branch)
+            Ybus, Yf, Yt = makeYbus(baseMVA, bus, branch)  # todo expansion with SSC
```
Comment (SteffenMeinecke) [inline pandapower/pypower/newtonpf.py:300]: this todo should have been addressed no, right?

File: pandapower/pypower/newtonpf.py [hunk 4]
```diff
@@ -306,8 +343,12 @@ def newtonpf(Ybus, Sbus, V0, ref, pv, pq, ppci, options, makeYbus=None):
             x_control_svc[svc_controllable] += dx[j6:j6a]
         if any_tcsc_controllable:
             x_control_tcsc[tcsc_controllable] += dx[j6a:j6b]
+        # if any_ssc_controllable:
+        #     x_control_ssc[ssc_controllable] += dx[j6b:j6c]
+        #     x_control_ssc[ssc_controllable] += dx[j6c:j6d]
+
         if tdpf:
-            T = T + dx[j7:][tdpf_lines]
+            T = T + dx[j8:][tdpf_lines]  # todo check here if it is still correct
```
Comment (SteffenMeinecke) [inline pandapower/pypower/newtonpf.py:351]: Is this todo checked already?

**Review by SteffenMeinecke — DISMISSED** [review]

File: pandapower/pypower/idx_ssc.py (hunk 1 above)
Comment (rbolgaryn) [reply pandapower/pypower/idx_ssc.py:5]: True, should be removed

File: tutorials/FACTS.ipynb
[the remark this answers is not in the document]
Comment (rbolgaryn) [reply tutorials/FACTS.ipynb:1]: SSC is already added to the tutorial FACTS.ipynb. However, there are warnings from numba for some reason, which must be cleaned up.

File: pandapower/pypower/newtonpf.py (hunk 3 above)
Comment (rbolgaryn) [reply pandapower/pypower/newtonpf.py:302]: clarified the TODO message. We will address it after completing the FACTS implementations.

File: pandapower/pf/create_jacobian_facts.py (hunk 2 above)
Comment (rbolgaryn) [reply pandapower/pf/create_jacobian_facts.py:188]: removed it

**Review by SteffenMeinecke — APPROVED** [review]

**Pull request merged**